File: 2779-maximum-beauty-of-an-array-after-applying-operation/2779-maximum-beauty-of-an-array-after-applying-operation.js
var maximumBeauty = function(nums, k) {
    const mx = Math.max(...nums);

    const vis = new Array(mx + 2).fill(0);

    for (const x of nums) {
        vis[Math.max(0, x - k)]++;
        vis[Math.min(x + k + 1, mx + 1)]--;
    }

    let ans = 0, cnt = 0;
    for (const x of vis) {
        cnt += x;
        ans = Math.max(ans, cnt);
    }

    return ans;
};

File: 2779-maximum-beauty-of-an-array-after-applying-operation/2779-maximum-beauty-of-an-array-after-applying-operation.java

class Solution {
    public int maximumBeauty(int[] nums, int k) {
        int mx = Arrays.stream(nums).max().getAsInt();

        int[] vis = new int[mx + 2];

        for (int x : nums) {
            vis[Math.max(0, x - k)]++;
            vis[Math.min(x + k + 1, mx + 1)]--;
        }

        int ans = 0, cnt = 0;
        for (int x : vis) {
            cnt += x;
            ans = Math.max(ans, cnt);
        }

        return ans;
    }
}
